table/table.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package table
+
+import (
+	"reflect"
+
+	"github.com/apache/iceberg-go"
+	"github.com/apache/iceberg-go/io"
+	"golang.org/x/exp/slices"
+)
+
+type Identifier = []string
+
+type Table struct {
+	identifier       Identifier
+	metadata         Metadata
+	metadataLocation string
+	fs               io.IO
+}
+
+func (t Table) Equals(other Table) bool {
+	return slices.Equal(t.identifier, other.identifier) &&
+		t.metadataLocation == other.metadataLocation &&
+		reflect.DeepEqual(t.metadata, other.metadata)
+}
+
+func (t Table) Identifier() Identifier   { return t.identifier }
+func (t Table) Metadata() Metadata       { return t.metadata }
+func (t Table) MetadataLocation() string { return t.metadataLocation }
+func (t Table) FS() io.IO                { return t.fs }
+
+func (t Table) Schema() *iceberg.Schema              { return t.metadata.CurrentSchema() }
+func (t Table) Spec() iceberg.PartitionSpec          { return t.metadata.PartitionSpec() }
+func (t Table) SortOrder() SortOrder                 { return t.metadata.SortOrder() }
+func (t Table) Properties() iceberg.Properties       { return t.metadata.Properties() }
+func (t Table) Location() string                     { return t.metadata.Location() }
+func (t Table) CurrentSnapshot() *Snapshot           { return t.metadata.CurrentSnapshot() }
+func (t Table) SnapshotByID(id int64) *Snapshot      { return t.metadata.SnapshotByID(id) }
+func (t Table) SnapshotByName(name string) *Snapshot { return t.metadata.SnapshotByName(name) }
```

Comment:
do we need to expose SnapshotByXyz for Table?

Reply:
I added these because pyiceberg had them exposed (https://github.com/apache/iceberg-python/blob/main/pyiceberg/table/__init__.py#L553), so I figured I should follow suit. pyiceberg also uses the singular for the methods too :smile:

Reply:
ok thanks for clarifying